### PR TITLE
extracted compute_clusters function

### PR DESCRIPTION
**Pull request by hechth**

### CODE REVIEW

File: R/adjust.time.R [hunk 1]
```diff
@@ -75,14 +75,68 @@ fill_missing_values <- function(orig.feature, this.feature) {
...
+    all,
+    number_of_samples = number_of_samples,
+    mz_tol_relative = mz_tol_relative,
+    rt_tol_relative = rt_tol_relative,
+    mz_tol_absolute = mz_tol_absolute,
+    do.plot = do.plot
+  )
+  rt_tol_relative <- all.ft$rt.tol
+
+  message("**** performing time correction ****")
```
Comment (xtrojak) [inline R/adjust.time.R:138]: This message is incorrect in the case of features alignment.

File: R/adjust.time.R [hunk 2]
```diff
@@ -131,58 +185,19 @@ adjust.time <- function(extracted_features,
...
-      i
-    )
-    extracted_features[[i]] <- features
-  }
+  extracted_features <- res$feature_tables
+  rt_tol_relative <- res$rt_tol_relative
+  mz_tol_relative <- res$mz_tol_relative
 
-  num.ftrs <- sapply(extracted_features, nrow)#as.vector(table(all.ft$sample_id))
+  num.ftrs <- sapply(extracted_features, nrow) # as.vector(table(all.ft$sample_id))
```
Comment (xtrojak) [inline R/adjust.time.R]:
```suggestion
  num.ftrs <- sapply(extracted_features, nrow)
```

File: R/feature.align.R [hunk 3]
```diff
@@ -179,85 +176,25 @@ feature.align <- function(features,
...
-        if (is.na(mz_tol_relative)) {
-            mz_tol_relative <- find.tol(
-                mz_values,
-                mz_max_diff = mz_max_diff,
-                do.plot = do.plot
-            )
-            if (length(mz_tol_relative) == 0) {
-                mz_tol_relative <- 1e-5
-                warning("Automatic tolerance finding failed, 10 ppm was assigned.
-                        May need to manually assign alignment mz tolerance level.")
```
Comment (xtrojak) [inline R/feature.align.R:198]: This is not reproduced in refactored version.

File: R/feature.align.R [hunk 4]
```diff
@@ -179,85 +176,25 @@ feature.align <- function(features,
...
-        message(paste("time tolerance level:", rt_tol_relative))
-
+        
+        all_table <- dplyr::bind_rows(res$feature_tables)
+        rt_tol_relative <- res$rt_tol_relative
+        mz_tol_relative <- res$mz_tol_relative
+        
         # create zero vectors of length number_of_samples + 4 ?
         aligned_features <- pk.times <- NULL
         mz_sd <- 0
```
Comment (xtrojak) [inline R/feature.align.R]:
```suggestion
```

**Review by xtrojak — APPROVED** [review]
That looks awesome, just minor suggestions. Good job with joining all vectors to single structure!

File: R/feature.align.R (hunk 3 above)
Comment (hechth) [reply R/feature.align.R:198]: See https://github.com/RECETOX/recetox-aplcms/pull/102/commits/7a9346adf0d18020bc15496997b5f744c5c37456

**Pull request merged**